### ISSUE TITLE
[Bug] macOS clang 13 build throw [-Werror,-Wdeprecated-copy]

### ISSUE THREAD

**Issue by awemvfx** [firebase/firebase-cpp-sdk#914]
### [REQUIRED] Please fill in the following fields:

  * open-source from this repo
  * Firebase C++ SDK version: 8.11.0
  * Problematic Firebase Component: firestore
  * Platform you are using the C++ SDK on: Mac
  * Platform you are targeting: desktop

### [REQUIRED] Please describe the issue here:

Build  **firestore** module from source with **Apple clang version 13.1.6 (clang-1316.0.21.2.3)** generate errors:

```
In file included from /Users/awe/qt_projects/firebase-cpp-sdk/build/external/src/firestore/Firestore/core/src/core/firestore_client.cc:34:
In file included from /Users/awe/qt_projects/firebase-cpp-sdk/build/external/src/firestore/Firestore/core/src/core/sync_engine.h:37:
In file included from /Users/awe/qt_projects/firebase-cpp-sdk/build/external/src/firestore/Firestore/core/src/remote/remote_store.h:29:
In file included from /Users/awe/qt_projects/firebase-cpp-sdk/build/external/src/firestore/Firestore/core/src/remote/datastore.h:26:
In file included from /Users/awe/qt_projects/firebase-cpp-sdk/build/external/src/firestore/Firestore/core/src/credentials/auth_token.h:22:
/Users/awe/qt_projects/firebase-cpp-sdk/build/external/src/firestore/Firestore/core/src/credentials/user.h:73:9: error: definition of implicit copy constructor for 'User' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]
  User& operator=(const User& other) = default;
        ^
/Users/awe/qt_projects/firebase-cpp-sdk/build/external/src/firestore/Firestore/core/src/core/firestore_client.cc:134:61: note: in implicit copy constructor for 'firebase::firestore::credentials::User' first required here
      shared_client->worker_queue_->Enqueue([shared_client, user, settings] {
                                                            ^
1 error generated.
make[3]: *** [external/src/firestore-build/Firestore/core/CMakeFiles/firestore_core.dir/src/core/firestore_client.cc.o] Error 1
make[3]: *** Waiting for unfinished jobs....
In file included from /Users/awe/qt_projects/firebase-cpp-sdk/build/external/src/firestore/Firestore/core/src/core/sync_engine.cc:17:
In file included from /Users/awe/qt_projects/firebase-cpp-sdk/build/external/src/firestore/Firestore/core/src/core/sync_engine.h:37:
In file included from /Users/awe/qt_projects/firebase-cpp-sdk/build/external/src/firestore/Firestore/core/src/remote/remote_store.h:29:
In file included from /Users/awe/qt_projects/firebase-cpp-sdk/build/external/src/firestore/Firestore/core/src/remote/datastore.h:26:
In file included from /Users/awe/qt_projects/firebase-cpp-sdk/build/external/src/firestore/Firestore/core/src/credentials/auth_token.h:22:
/Users/awe/qt_projects/firebase-cpp-sdk/build/external/src/firestore/Firestore/core/src/credentials/user.h:73:9: error: definition of implicit copy constructor for 'User' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]
  User& operator=(const User& other) = default;
        ^
/Users/awe/qt_projects/firebase-cpp-sdk/build/external/src/firestore/Firestore/core/src/core/sync_engine.cc:95:7: note: in implicit copy constructor for 'firebase::firestore::credentials::User' first required here
      current_user_(initial_user),
      ^
1 error generated.
make[3]: *** [external/src/firestore-build/Firestore/core/CMakeFiles/firestore_core.dir/src/core/sync_engine.cc.o] Error 1
make[2]: *** [external/src/firestore-build/Firestore/core/CMakeFiles/firestore_core.dir/all] Error 2
make[1]: *** [firestore/CMakeFiles/firebase_firestore.dir/rule] Error 2
make: *** [firebase_firestore] Error 2
```

**Comment by dconeybe** [comment]
Thank you for the bug report. This appears to be a problem in the [firebase-ios-sdk](https://github.com/firebase/firebase-ios-sdk) repository.

Could you please try building that repository directly and see what happens? If you get the same/similar errors, please open a bug report there.

FWIW I ran into a similar build error with Xcode 13.3.1. Downgrading to Xcode 13.2.1 fixed the build error.

Also, please run `clang --version` and include the output from that command in your bug report, as well as the version of Xcode that you have installed.

**Comment by awemvfx** [comment]
Xcode 13.3.1
Apple clang version 13.1.6 (clang-1316.0.21.2.3)
Target: x86_64-apple-darwin21.4.0
Thread model: posix

Building firebase-ios-sdk directly throw this same error. I created a new report there.
https://github.com/firebase/firebase-ios-sdk/issues/9702

**Comment by dconeybe** [comment]
Thank you. I'm going to close this issue and use the other one you created to fix the problem at the source.

**Comment by dconeybe** [comment]
The fix has been incorporated into version 9.0.0, released May 10, 2022.
